oteapi/strategies/mapping/mapping.py

```diff
@@ -0,0 +1,63 @@
+"""Mapping filter strategy."""
+# pylint: disable=unused-argument
+from typing import TYPE_CHECKING
+
+from pydantic.dataclasses import Field, dataclass
+
+from oteapi.models import MappingConfig, SessionUpdate
+
+if TYPE_CHECKING:  # pragma: no cover
+    from typing import Any, Dict, Optional
+
+
+class MappingStrategyConfig(MappingConfig):
+    """Strategy-specific mapping config."""
+
+    mappingType: str = Field(
+        "triples",
+        description="Mapping type.  Normally this is just 'mapping'.",
+    )
+
+
+@dataclass
+class MappingStrategy:
+    """Strategy for a mapping.
+
+    The mapping strategy simply adds more prefixes and triples to the
+    `prefixes` and `triples` fields in the session.
+
+    Nothing is returned to avoid deleting existing mappings.
+
+    **Registers strategies**:
+
+    - `("mappingType", "triples")`
+
+    """
+
+    config: MappingStrategyConfig
+
+    def initialize(self, session: "Optional[Dict[str, Any]]" = None) -> SessionUpdate:
+        """Initialize strategy."""
+        if session is None:
+            return_session = True
+            session = {}
+        else:
+            return_session = False
+
+        if not "prefixes" in session:
+            session["prefixes"] = {}
+
+        if not "triples" in session:
+            session["triples"] = []
+
+        session["prefixes"].update(self.config.prefixes)
+        session["triples"].extend(self.config.triples)
+
+        if return_session:
+            return SessionUpdate(**session)
+        else:
+            return SessionUpdate()
```

Comment:
I don't understand your logic at all in this method.
This is what I suggest to have here instead:

```suggestion
        session = session or {}

        if not "prefixes" in session:
            session["prefixes"] = {}

        if not "triples" in session:
            session["triples"] = []

        session["prefixes"].update(self.config.prefixes)
        session["triples"].extend(self.config.triples)

        return SessionUpdate(**session)
```